pages/book/structs-and-messages.mdx

```diff
@@ -198,9 +198,43 @@ fun conversionFun() {
 <Callout>
 
   See those extension functions in the Reference:\
-  [`Struct.toCell(){:tact}`](/ref/api-cells#structtocell)\
-  [`Message.toCell(){:tact}`](/ref/api-cells#messagetocell)
+  [`Struct.toCell(){:tact}`](/ref/core-cells#structtocell)\
+  [`Message.toCell(){:tact}`](/ref/core-cells#messagetocell)
 
 </Callout>
 
+### Obtain from a `Cell` or `Slice`, `.fromCell()` and `.fromSlice()` [#fromcellslice]
+
+Instead of manually parsing a [`Cell{:tact}`][p] or [`Slice{:tact}`][p] via the series of relevant `.loadSomething(){:tact}` function calls, one can use `.fromCell(){:tact}` and `.fromSlice(){:tact}` [extension functions](/book/functions#extension-function) for converting the provided [`Cell{:tact}`][p] or [`Slice{:tact}`][p] into the needed [Struct](#structs) or [Message](#messages).
+
+Those extension functions only attempt to parse a [`Cell{:tact}`][p] or [`Slice{:tact}`][p] according to the structure of your [Struct](#structs) or [Message](#messages). In case layouts don't match, various exceptions may be thrown — make sure to wrap your code in [`try...catch{:tact}`](/book/statements#try-catch) blocks to prevent unexpected results.
+
```

Comment:
would be nice to add the following laws hold:

- for any `struct` of type `Struct` (which is a struct or a message):

```
Struct.fromCell(struct.toCell()) == struct
```
and

- for any `cell` and type `Struct` (which is a struct or a message), if `Struct.fromCell(cell)` does not throw, then

```
Struct.fromCell(cell).toCell() == cell
```

and the same goes for `toSlice` / `fromSlice`.

Reply:
@anton-trunov Those are nice, but:

* `toSlice(){}` doesn't exist. One can do `Struct.toCell().asSlice()` though
* Equality operator `==` cannot compare Structs and Messages directly. One would need to compare them as Cells (at the moment)

Reply:
Good point about `toSlice()` -- we should probably add it, for consistency reasons, e.g. `StringBuilder` has it.

The equality `==` operator is meant here in the mathematical sense, it's an instance of equational reasoning which is used commonly in functional programming and `=` seemed to be a bit inappropriate given its imperative meaning. We might want to use `===`, add an explanation here, or use LaTeX and possible quantifiers: `∀ s: Struct, Struct.fromCell(s.toCell()) = s`